Sources/TuistKit/Services/ShareService.swift

```diff
@@ -131,29 +138,30 @@ struct ShareService {
             )
         }
 
-        if !apps.isEmpty, apps.allSatisfy({ $0.hasSuffix(".app") }) {
-            let appPaths = try apps.map {
-                try AbsolutePath(
-                    validating: $0,
-                    relativeTo: fileHandler.currentPath
+        let appPaths = try await apps.concurrentMap {
+            try AbsolutePath(
+                validating: $0,
+                relativeTo: try await fileSystem.currentWorkingDirectory()
+            )
+        }
+        if !apps.isEmpty, try await appPaths
+            .concurrentMap({ try await fileSystem.exists($0) })
+            .allSatisfy({ $0 })
+        {
+            if appPaths.contains(where: { $0.extension == "ipa" }) {
+                try await shareIPA(
+                    appPaths,
+                    fullHandle: fullHandle,
+                    serverURL: serverURL
+                )
+            } else {
+                try await shareAppBundles(
+                    appPaths,
+                    fullHandle: fullHandle,
+                    serverURL: serverURL
                 )
             }
 
-            let appBundles = try await appPaths.concurrentMap {
-                try await appBundleLoader.load($0)
-            }
-
-            let appNames = appBundles.map(\.infoPlist.name).uniqued()
-            guard appNames.count == 1,
-                  let appName = appNames.first else { throw ShareServiceError.multipleAppsSpecified(appNames) }
-
-            let preview = try await previewsUploadService.uploadPreviews(
-                displayName: appName,
-                previewPaths: appPaths,
-                fullHandle: fullHandle,
-                serverURL: serverURL
-            )
-            logger.notice("\(appName) uploaded – share it with others using the following link: \(preview.url.absoluteString)")
         } else if try await manifestLoader.hasRootManifest(at: path) {
             guard apps.count < 2 else { throw ShareServiceError.multipleAppsSpecified(apps) }
```

Comment:
This check is also in `shareIPA`. Shouldn't we have it once at the service level? Also, what do you think if we surface that in the help so that users don't try to pass multiple apps?

Reply:
We should keep it here because when you share individual app bundles with their paths, you can specify more of them. So, this check can't be done earlier.